contracts/REXTwoWayUniswapMarket.sol

```diff
@@ -0,0 +1,614 @@
+                address(inputTokenA),
+                2**256 - 1
+            );
+        }
+
+        if (inputTokenBUnderlying == address(0)) {
+            // inputTokenB is supertoken, approve swap router for it
+            inputTokenBUnderlying = address(inputTokenB);
+        } else {
+            // otherwise approve underlying for upgrade
+            ERC20(inputTokenBUnderlying).safeIncreaseAllowance(
+                address(inputTokenB),
+                2**256 - 1
+            );
+        }
+
+        ERC20(inputTokenAUnderlying).safeIncreaseAllowance(
+            address(router),
+            2**256 - 1
+        );
+        ERC20(inputTokenBUnderlying).safeIncreaseAllowance(
+            address(router),
+            2**256 - 1
+        );
+
+        market.lastDistributionAt = block.timestamp;
+    }
+
+    function initializeUniswap(
+        ISwapRouter02 _uniswapRouter,
+        address[] memory _uniswapPath,
+        uint24[] memory _poolFees
+    ) external onlyOwner {
+        router = _uniswapRouter;
+        uniswapPath = _uniswapPath;
+        poolFees = _poolFees;
+    }
+
+    function initializeSubsidies(
+        uint256 _emissionRate,
+        ISuperToken _subsidyToken
+    ) public onlyOwner {
+        subsidyToken = _subsidyToken;
+        require(
+            address(market.outputPools[SUBSIDYA_INDEX].token) == address(0) &&
+                address(market.outputPools[SUBSIDYB_INDEX].token) == address(0),
+            "already initialized"
+        );
+        addOutputPool(
+            _subsidyToken,
+            0,
+            _emissionRate,
+            77,
+            market.outputPools[OUTPUTB_INDEX].shareScaler
+        );
+        addOutputPool(
+            _subsidyToken,
+            0,
+            _emissionRate,
+            77,
+            market.outputPools[OUTPUTA_INDEX].shareScaler
+        );
+        lastDistributionTokenAAt = block.timestamp;
+        lastDistributionTokenBAt = block.timestamp;
+        // Does not need to add subsidy token to outputPoolIndicies
+        // since these pools are hardcoded
+    }
+
+    function addOutputPool(
+        ISuperToken _token,
+        uint128 _feeRate,
+        uint256 _emissionRate,
+        uint256 _requestId,
+        uint128 _shareScaler
+    ) public override onlyOwner {
+        // Only Allow 4 output pools, this overrides the block in REXMarket
+        // where there can't be two output pools of the same token
+        require(market.numOutputPools < 4, "too many pools");
+
+        OutputPool memory _newPool = OutputPool(
+            _token,
+            _feeRate,
+            _emissionRate,
+            _shareScaler
+        );
+        market.outputPools[market.numOutputPools] = _newPool;
+        market.outputPoolIndicies[_token] = market.numOutputPools;
+        _createIndex(market.numOutputPools, _token);
+        market.numOutputPools++;
+        OracleInfo memory _newOracle = OracleInfo(_requestId, 0, 0);
```

Comment:
This will need to change, you probably don't need to override this method at all in fact bc its implemented correctly for this in `REXMarket`